boards/arm/96b_carbon/96b_carbon_defconfig

```diff
@@ -50,3 +50,8 @@ CONFIG_BT_SPI_IRQ_PIN=1
 CONFIG_BT_SPI_RESET_DEV_NAME="GPIOB"
 CONFIG_BT_SPI_RESET_PIN=4
 CONFIG_BT_SPI_MAX_CLK_FREQ=2000000
+
+# BT is the only onboard network iface, so use it for IP networking
+# if it's enabled
+CONFIG_NET_L2_BT=y
+CONFIG_NET_L2_BT_ZEP1656=y
```

Comment:
~~~
$ grep -r CONFIG_NET_L2_BT_ZEP1656=y * | grep prj
bluetooth/ipsp/prj_dbg.conf:CONFIG_NET_L2_BT_ZEP1656=y
bluetooth/ipsp/prj.conf:CONFIG_NET_L2_BT_ZEP1656=y
net/http_server/prj_bt.conf:CONFIG_NET_L2_BT_ZEP1656=y
net/zoap_server/prj_bt.conf:CONFIG_NET_L2_BT_ZEP1656=y
net/zoap_client/prj_bt.conf:CONFIG_NET_L2_BT_ZEP1656=y
net/https_client/prj_bt.conf:CONFIG_NET_L2_BT_ZEP1656=y
net/echo_server/prj_bt.conf:CONFIG_NET_L2_BT_ZEP1656=y
net/sockets/echo/prj.conf.1:CONFIG_NET_L2_BT_ZEP1656=y
net/sockets/echo/prj.conf:#CONFIG_NET_L2_BT_ZEP1656=y
net/sockets/echo_ipv6/prj.conf:CONFIG_NET_L2_BT_ZEP1656=y
net/http_client/prj_bt.conf:CONFIG_NET_L2_BT_ZEP1656=y
net/zperf/prj_bt.conf:CONFIG_NET_L2_BT_ZEP1656=y
~~~

I told in #1151 that it's a moot point - **all** BT-related networking sample enable it anyway. By moving "y" value to Kconfig, we'll actually make it easier to disable it - now for bleeding-edge parties and later at all.

In either case, 96b_carbon is intended for real people who don't run the latest kernels, so here it's at the right place (subject for ack'ing by @mbolivar and @rsalveti. But I can't imagine it to be otherwise - I was asked to write a blog post on sockets on 96b_carbon, and of course I can start it with "We worked on 96b_carbon upstreaming in Zephyr for a year, and guess what - we still aren't there, you still need to do heavy patching to use it.", but I do my work to avoid that.)